Add Quarto Space docs (#1226)
huggingface/hub-docs: docs/hub/spaces-sdks-docker-quarto.md

# Quarto on Spaces

> Quarto is a powerful tool for creating reproducible scientific and technical documents.

Reviewer: ```suggestion
[Quarto](https://quarto.org/) is a powerful tool for creating reproducible scientific and technical documents.
```

> With Quarto you can interleave markdown and code chunks to generate pdfs, websites, presentations and more.

Reviewer: ```suggestion
With Quarto, you can interleave markdown and code chunks to generate pdfs, websites, presentations, and more.
```

### Getting Started with Quarto on Spaces

1. Create the Quarto space, and clone it locally

> <a  href="https://huggingface.co/new-space?template=posit/quarto-template"> <img src="https://huggingface.co/datasets/huggingface/badges/raw/main/deploy-to-spaces-lg.svg"/> </a>

Reviewer: I would split this into two steps, and I think you should add an example code snippet on how to clone the repo

Reviewer: Actually, cloning locally is not a strict requirement no? People can do the development entirely in the UI. I would suggest to split this into

* Creating and setting up the repo, doing changes in the Space by directly modifying `/src`
* Developing locally (install quarto, VS Code extension)
* Learn more about  Quarto

Author: I suppose that's true, but working on a quarto site in the browser without being able to preview the site. Happy to change the organization though.

Reviewer: I think mentioning, even if briefly, the in-browser setup could be ok!

2. **Install Quarto**: In order to render your Quarto site without Docker, we recommend installing Quarto by following the instructions on the [official Quarto website](https://quarto.org/docs/get-started/).

> 3. **Install Quarto VS Code extension** the [Quarto VS Code Extention](https://quarto.org/docs/tools/vscode.html) includes a number of productivity tools including YAML Autocomplete, a preview button, and a visual editor. Quarto works great with VS Code, but the extension does make it easier to get the most out of Quarto.

Reviewer: ```suggestion
3. **Install Quarto VS Code extension** the [Quarto VS Code Extention](https://quarto.org/docs/tools/vscode.html) includes a number of productivity tools, including YAML Autocomplete, a preview button, and a visual editor. Quarto works great with VS Code, but the extension does make it easier to get the most out of Quarto.
```

> 4. **Edit the site** The website files are contained in the `src` directory, and the site navigation is defined in `src/_quarto.yml`, try editing these files and either clicking the "Preview" button in VS Code, or calling `quarto preview src` from the command line.

Reviewer: ```suggestion
4. **Edit the site** The website files are contained in the `src` directory, and the site navigation is defined in `src/_quarto.yml`, try editing these files and either clicking the "Preview" button in VS Code or calling `quarto preview src` from the command line.
```

> 5. **Learn more about Quarto** You can do a lot of things with Quarto, and they are all documented on the [Quarto Website](https://quarto.org/guide/). In particular you may be interested in:

Reviewer: ```suggestion
5. **Learn more about Quarto** You can do a lot of things with Quarto, and they are all documented on the [Quarto Website](https://quarto.org/guide/). In particular, you may be interested in:
```

# Quarto on Spaces

[Quarto](https://quarto.org/) is a powerful tool for creating reproducible scientific and technical documents.
With Quarto, you can interleave markdown and code chunks to generate pdfs, websites, presentations, and more.

## Getting Started with Quarto on Spaces

### Edit the site on Hugging Face

1. Create the Quarto space

> <a  href="https://huggingface.co/new-space?template=posit/quarto-template"> <img src="https://huggingface.co/datasets/huggingface/badges/raw/main/deploy-to-spaces-lg.svg"/> </a>

Reviewer 2: Maybe mention that a CPU Space is enough?

2. Edit in the browser

You can make changes to the quarto site directly in Hugging Face.
There are three main types of files which you will need to edit in order to make changes:

- The `src/_quarto.yml` file contains the site navigation, and is the best place to start if you want to change the structure of the site.
- The notebook and `.qmd` files in the `src` directory contain the actual content of the web pages.

> - The `Dockerfile` contains the instructions for building the site. For example if you want to change the version of Quarto which the site uses, you should update the Dockerfile.

Reviewer 2: ```suggestion
- The `Dockerfile` contains the instructions for building the site. For example, if you want to change the version of Quarto which the site uses, you should update the Dockerfile.
```

3. Commit your changes and build the site

### Local development

Editing the site on Hugging Face is a convenient way to make small changes, but most of the time you will want to develop and preview the site locally before deploying.
This will allow you to view your changes more quickly, and to use the full power of the Quarto development environment.

1. Create the Quarto space as above

2. Clone the Repo

> To clone the space locally run the following command, make sure to replace `<YOUR_HF_USER>` with your Hugging Face username.

Reviewer 2: ```suggestion
To clone the space locally run the following command, making sure to replace `<YOUR_HF_USER>` with your Hugging Face username and `quarto-template` with the name you chose for your space.
```

> To clone the space locally run the following command, make sure to replace `<YOUR_HF_USER>` with your Hugging Face username.

Reviewer 2: They might need to be logged in if the Space is private.

```
git clone https://huggingface.co/spaces/<YOUR_HF_USER>/quarto-template
```

3. Install Quarto

In order to render your Quarto site without Docker, we recommend installing Quarto by following the instructions on the [official Quarto website](https://quarto.org/docs/get-started/).

3. Install Quarto VS Code extension

> The [Quarto VS Code Extention](https://quarto.org/docs/tools/vscode.html) includes a number of productivity tools, including YAML Autocomplete, a preview button, and a visual editor. Quarto works great with VS Code, but the extension does make it easier to get the most out of Quarto.

Reviewer 2: did you mean to say `without` VS Code?

Author: Whoops, yes I did

4. Edit the site
   The website files are contained in the `src` directory, and the site navigation is defined in `src/_quarto.yml`, try editing these files and either clicking the "Preview" button in VS Code or calling `quarto preview src` from the command line.

5. Commit your changes and push

>    Your site will rebuild whenever you push new commits to main branch of the space repository

Reviewer 2: ```suggestion
   Your site will rebuild whenever you push new commits to the `main` branch of your space repository.
```

### Learn about Quarto

You can do a lot of things with Quarto, and they are all documented on the [Quarto Website](https://quarto.org/guide/). In particular, you may be interested in:

- All about building [websites](https://quarto.org/docs/websites/)
- Building Static [Dashboards](https://quarto.org/docs/dashboards/)
- How to write [books](https://quarto.org/docs/books/index.html) and [manuscripts](https://quarto.org/docs/manuscripts/)
- Reproducible [presentations](https://quarto.org/docs/manuscripts/)

> - Including [Observable](https://quarto.org/docs/interactive/ojs/) or [Shiny](https://quarto.org/docs/interactive/shiny/) applications in your Quarto site

Reviewer 2: Nice!
